src/carousel/index.ts

```diff
@@ -0,0 +1,234 @@
+    this.resetCurrentSlideNum(from, to, length);
+    this.animationEndCallback = null;
+  }
+
+  resetCurrentSlideNum = (from: number, to: number, length: number) => {
+    let currentSlide;
+    if (from < to) {
+      currentSlide = to % length;
+    } else {
+      currentSlide = (to % length + length) % length;
+    }
+    this.set({ currentSlide });
+  }
+
+  disableTransition = () => {
+    this.refs.track.removeEventListener('transitionend', this.disableTransition);
+    this.set({ transitioning: false });
+  }
+
+  trackStyle = () => {
+    let transitionStyles = {};
+    if (this.state.transitioning && this.state.settings.speed) {
+      const speed = this.state.settings.speed;
+      const transition = speed + 'ms ' + 'ease';
+      transitionStyles = {
+        '-webkit-transition': transition,
+        transition,
+        '-ms-transition': transition
+      };
+    }
+
+    return { ...this.getStaticTrackStyle(), ...transitionStyles };
+  }
+
+  getStaticTrackStyle = () => {
+    const slideWidth = this.getSlideWidth();
+    const slideCount = this.props.items.length;
+    const slidesToShow = this.state.settings.slidesToShow;
+    const trackWidth = (slideCount + 4 * slidesToShow - 2) * slideWidth;
+    const pos = this.calculatePosition(this.state.currentSlide, slideWidth, slidesToShow);
+    const transform = `translateX(-${pos}px)`;
+    const transformStyles = {
+      transform,
+      '-webkit-transform': transform,
+      '-ms-transform': transform
+    };
+
+    return { width: `${trackWidth}px`, ...transformStyles };
+  }
+
+  getSlideWidth = () => {
+    const visibleWidth = this.refs.wrapper.offsetWidth;
+
+    if (visibleWidth) {
+      return visibleWidth / this.state.settings.slidesToShow;
+    }
+  }
+
+  // tslint:disable-next-line:max-line-length
+  calculatePosition = (currentSlide: number, moveDistance: number, slidesToShow: number): number => (currentSlide + slidesToShow * 2 - 1) * moveDistance;
+
+  onTouchStart = (e: TouchEvent & Carousel.Event) => {
+    e.stopPropagation();
+
+    this.state.touchObject = {
+      startX: e.touches[0].pageX,
+      startY: e.touches[0].pageY
+    };
+
+    e.target.addEventListener('touchend', this.onTouchEnd);
+  }
+
+  onTouchEnd = (e: TouchEvent & Carousel.Event) => {
+    const curX = e.changedTouches[0].pageX;
+    const curY = e.changedTouches[0].pageY;
+    // swipe distance needs to be more than 20
+    if (Math.abs(curX - this.state.touchObject.startX) > 20) {
+      Carousel.isSwipeToNext({ ...this.state.touchObject }, curX, curY) ? this.moveNext() : this.movePrevious();
+    }
+
+    e.target.removeEventListener('touchstart', this.onTouchStart);
+    e.target.removeEventListener('touchend', this.onTouchEnd);
+  }
+
+  static isSwipeToNext = (touchObj: Carousel.TouchObject, curX: number, curY: number): boolean => {
+    const xDist = touchObj.startX - curX;
+    const yDist = touchObj.startY - curY;
+
+    const r = Math.atan2(yDist, xDist);
```

Comment:
I think the above variables could be inlined.

Reply:
I think it would be better inlined, we don't benefit from having a `xDist` variable as it's only used once, and the difference between two "X" values or two "Y" values it's clear what the intention is when subtracting one from the other

Reply:
also inline this into the next line too
```js
let swipeAngle = Math.round(Math.atan2(startY - curY, startX - curX) * 180 / Math.PI);
```